gpage.h

```diff
@@ -222,16 +235,19 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		//  Use std::less<> to compare (possibly unrelated) pointers portably
+		auto const cmp = std::less<>{};
+		auto const ext = extent();
+		return !cmp(p, ext.data()) && cmp(p, ext.data() + ext.size());
```

Comment:
Thanks, I hadn't noticed that. I'm glad it's being resolved after years of this code either being undefined (originally in C) or unspecified.